packages/@dcl/sdk/cli/commands/build/helpers.ts

```diff
@@ -0,0 +1,101 @@
+import { resolve } from 'path'
+
+// import spinner from '../../utils/spinner'
+import { exists, readdir, readFile } from '../../utils/fs'
+import { exec } from '../../utils/exec'
+import { Dict, hasPrimitiveKeys } from '../../utils/object'
+
+/**
+ * Required project dirs/files
+ */
+export const REQUIRED_FILES = {
+  PACKAGE_JSON: 'package.json',
+  TS_CONFIG_JSON: 'tsconfig.json',
+  SCENE_JSON: 'scene.json'
+}
+
+/**
+ * Required "package.json" structure
+ */
+export const REQUIRED_PACKAGE_JSON = {
+  devDependencies: {
+    '@dcl/sdk': ['latest']
+  }
+}
+
+/**
+ * Returns the required files for a project
+ */
+export const getProjectStructure = () => Object.values(REQUIRED_FILES)
+
+/**
+ * Returns true if the project follows a valid scene structure
+ */
+export const validateProjectStructure = async (
+  dir: string,
+  fileList: string[]
+): Promise<boolean> => {
+  const files = await readdir(dir)
+  const requiredFiles = new Set(fileList)
+
+  for (const file of files) {
```

Comment:
wont be better to iterate the `requiredFiles` instead of all the files of the project ? 
If you have only const requiredFiles = new Set('package.json'), you iterate this only once instead of checking every file if its equal to package.json

Reply:
it would be actually the same, creating the Set from `await readdir(dir)` implies iterating all the files. I don't think there is a way to avoid that.

Reply:
in fact, I think an improvement would be to just early-return in case all the required files are found before iterating over all the files in the directory. That way we improve the "best-case"  scenario. The worst case sceneario will remain the same (the last required file is the last file on the files array)

Reply:
pushed this:
```typescript
for (let i = 0; i < files.length && requiredFiles.size > 0; i++) {
  if (requiredFiles.has(files[i])) {
    requiredFiles.delete(files[i])
  }
}
```